app/controllers/csv_preview_controller.rb

```diff
@@ -25,35 +23,20 @@ def show
       redirect_to(parent_document_uri, status: :see_other, allow_other_host: true) and return
     end
 
-    original_error = nil
-    row_sep = :auto
-
-    begin
-      csv_preview = CSV.parse(media_download_truncated, encoding: encoding(media_download_truncated), headers: true, row_sep:)
-    rescue CSV::MalformedCSVError => e
-      if original_error.nil?
-        original_error = e
-        row_sep = "\r\n"
-        retry
-      else
-        render :malformed_csv and return
-      end
-    end
-
-    @csv_rows = csv_preview.to_a.map do |row|
-      row.map { |column|
-        {
-          text: column&.encode("UTF-8"),
-        }
-      }.take(MAXIMUM_COLUMNS)
-    end
+    @csv_rows = CsvPreviewService
+      .new(GdsApi.asset_manager.media(params[:id], params[:filename]).body)
+      .csv_rows
   end
 
+private
+
   def access_limited
-    render :access_limited, status: :forbidden and return
+    render :access_limited, status: :forbidden
   end
 
-private
+  def malformed_csv
+    render :malformed_csv
```

Comment:
How come we no longer need the `and return` ?

Reply:
As there's no controller test, I can't tell if this is an issue or not.

Reply:
return returns from the function, so it's redundant. What do you think return does in this code?

The controller has originally only had the integration test. I can add tests for that to it, or I can create a functional test and add it there. As this task was a refactoring task and not "write missing tests" task, I've only added tests which relate to the moved code, but I can add the missing controller tests, if that's your preference.

Reply:
I was thinking without the return we might get https://api.rubyonrails.org/classes/AbstractController/DoubleRenderError.html 

But if that's not then happy to lose it. Thanks for adding the test.

Reply:
I've added the test for malformed csv.

Reply:
https://github.com/alphagov/frontend/blob/8c08ee54269049bbf530846e1e4f8cf6e632073e/test/integration/csv_preview_test.rb#L112